modules/core/src/parallel.cpp

```diff
@@ -77,6 +84,14 @@
     #define HAVE_CONCURRENCY
 #endif
 
+#if (defined __linux__ || defined __GLIBC__ || defined __HAIKU__ || defined __EMSCRIPTEN__) && !defined __ANDROID__
+   #define CV_LINUX_GROUPS
+#endif
+
+#if (defined __linux__ || defined __GLIBC__ || defined __HAIKU__) && !defined __ANDROID__
```

Comment:
HAIKU is not Linux distribution, it's OSS BeOS implementation with it's own kernel and it does not support CGroups. GLIBC is not requirement for it too. I propose to stay with condition `#if (defined __linux__ && !defined __ANDROID__)`

Reply:
Yeah, I think I'll restructure the way we handle Linux, probably just split up the Linux group as there are many uncommon blocks